mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -165,35 +169,73 @@ class MDocUpdater : MDocCommand
 
 	HashSet<string> forwardedTypes = new HashSet<string> ();
 
-	public static string droppedNamespace = string.Empty;
+	public static DroppedAssemblyInfo.List droppedAssemblyList = new DroppedAssemblyInfo.List ();
+	public static string currentDroppedNamespace = string.Empty;
```

Comment:
This is very good feedback ... I only added the static state originally due to lack of time to do a bigger refactor, and this data needed to be referenced from a number of parts of the codebase. That being said, I didn't add this lightly, I'm aware of the limitations that come with adding static mutable state ... in fact, one of the things that I'm planning on doing is running mdoc through Xamarin's new profiler with an eye towards making it generally faster. Multithreading it is the obvious candidate to do that (other than general code/algorithm improvement) and this was first on my list to refactor out.

Is it cool if we leave the mutable state refactor until a later date? or do you want it resolved before merging this code in?